src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -74,6 +91,27 @@ def map_alternatives_items
       end
     end
 
+    def filtered_alternatives
+      alternatives = @alternatives_list.select { |a| Regexp.new(@search).match(a.name) }
+      if @multi_choice_only
+        alternatives.each_with_index.select { |v, _i| v.choices.length > 1 }
+      else
+        alternatives.each_with_index
+      end
+    end
+
+    def filters
+      VBox(
+        InputField(Id(:search), Opt(:notify), _("Search by name"), @search),
+        CheckBox(
+          Id(:multi_choice_only),
+          Opt(:notify),
+          _("Filter alternatives with one choice"),
```

Comment:
Maybe better to have text like "Show only alternatives with more then one choice"?